### PR TITLE
Add a test for javascript initiated video auto play

### PR DESCRIPTION
**Pull request by jraoult**
#### What's this PR do?

Adds a test that checks if a video can be started by javascript without user intervention (the JS call is not the consequence of a user initiated event)

#### Related issue
#1505

### CODE REVIEW

**Comment by jraoult** [comment]
@patrickkettner This is definitely not a PR ready for merging but I wanted to check with you the validity of this approach.

Since `video/autoplay` and `video/javascriptautoplay` differ only by the way they start the video I extracted to code to a parametrizable builder that returns the test itself.

Another thing, I forked from `master` (214f02af7b6e030de5d6422133c36975f74929bb) but I couldn't manage to get `grunt build` working so I couldn't run the tests. Where can I find a working version of the build script?

**Comment by patrickkettner** [comment]
@jraoult awesome work!

grunt build should be working - in what way is it not?

**Comment by jraoult** [comment]
@patrickkettner if think this doesn't work since in the dependencies I have `build` is synchronous and returns a value instead of accepting a callback:
```
modernizr.build(config, function(output) {
  grunt.file.write(dest, output);
});
```

The result is that there is no `./dist/modernizr-build.js` file generated

**Comment by patrickkettner** [comment]
actually, total screw up on my part. updating off master will give you a working version

**Comment by jraoult** [comment]
@patrickkettner thanks, it works now :-)

**Comment by jraoult** [comment]
@patrickkettner the results of my tests are disappointing because it seems that this test was not really required after all.  I can definitely confirm that autoplay (both by attribute and JS) passes for FF on Android and none of them pass for Chrome. So it seems that as of today autoplay triggered either by attribute or JS is treated the same way by browsers. I think I've been deceived by false negative for `videoautoplay` when I was prototyping this test earlier.

Bottom line we are testing two different use cases but the results are the same, does that make them redundant?

**Comment by patrickkettner** [comment]
yeah, if there is no browser that does one but not the other, then the
detect doesn't really detect anything.

That being said, I would urge you to triple check whether or not your
original findings were correct. Id hate to loose a perfectly good detect

thanks again, regardless!

**Comment by jraoult** [comment]
@patrickkettner no worries :-) yes, I'll check more thoroughly and I'll keep you posted

**Comment by patrickkettner** [comment]
closing in the mean time, please do ping if your results turn out otherwise.

cheers!

**Comment by jraoult** [comment]
Hey @patrickkettner, I found this issue in the FF bug tracker: https://bugzilla.mozilla.org/show_bug.cgi?id=659285

So it appears that it is possible in FF to disable "autoplay" as an attribute but still allow `play()` to be called in JS. This scenario would make my PR relevant after all :-)
